scheds/rust/scx_layered/src/main.rs

```diff
@@ -1376,38 +1525,89 @@ impl<'a> Scheduler<'a> {
+                calc_frac(stats.layer_loads[lidx], stats.total_load)
             );
-            info!(
-                "  {:<width$}  cpus={:3} [{:3},{:3}] {}",
-                "",
-                layer.nr_cpus,
-                self.nr_layer_cpus_min_max[lidx].0,
-                self.nr_layer_cpus_min_max[lidx].1,
-                format_bitvec(&layer.cpus),
-                width = header_width
+            let layer_tasks = layer_stat_set!(layer_tasks, stats.nr_layer_tasks[lidx] as i64);
+            let layer_total = layer_stat_set!(layer_total, ltotal as i64);
+            let layer_local =
+                layer_stat_set!(layer_local, lstat_pct(bpf_intf::layer_stat_idx_LSTAT_LOCAL));
+            let layer_open_idle = layer_stat_set!(
+                layer_open_idle,
+                lstat_pct(bpf_intf::layer_stat_idx_LSTAT_OPEN_IDLE)
+            );
+            let layer_preempt = layer_stat_set!(
+                layer_preempt,
+                lstat_pct(bpf_intf::layer_stat_idx_LSTAT_PREEMPT)
+            );
+            let layer_affn_viol = layer_stat_set!(
+                layer_affn_viol,
+                lstat_pct(bpf_intf::layer_stat_idx_LSTAT_AFFN_VIOL)
             );
+            let layer_cur_nr_cpus = layer_stat_set!(layer_cur_nr_cpus, layer.nr_cpus as i64);
+            let layer_min_nr_cpus =
+                layer_stat_set!(layer_min_nr_cpus, self.nr_layer_cpus_min_max[lidx].0 as i64);
+            let layer_max_nr_cpus =
+                layer_stat_set!(layer_max_nr_cpus, self.nr_layer_cpus_min_max[lidx].1 as i64);
+            if !self.open_metrics_format {
+                info!(
+                    "  {:<width$}: util/frac={:7.1}/{:5.1} load/frac={:9.1}:{:5.1} tasks={:6}",
+                    spec.name,
+                    layer_util.get(),
+                    layer_util_frac.get(),
+                    layer_load.get(),
+                    layer_load_frac.get(),
+                    layer_tasks.get(),
+                    width = header_width,
+                );
+                info!(
+                    "  {:<width$}  tot={:7} local={:5.2} open_idle={:5.2} preempt={:5.2} affn_viol={:5.2}",
+                    "",
+                    layer_total.get(),
+                    layer_local.get(),
+                    layer_open_idle.get(),
+                    layer_preempt.get(),
+                    layer_affn_viol.get(),
+                    width = header_width,
+                );
+                info!(
+                    "  {:<width$}  cpus={:3} [{:3},{:3}] {}",
+                    "",
+                    layer_cur_nr_cpus.get(),
+                    layer_min_nr_cpus.get(),
+                    layer_max_nr_cpus.get(),
+                    format_bitvec(&layer.cpus),
+                    width = header_width
+                );
+            }
             self.nr_layer_cpus_min_max[lidx] = (layer.nr_cpus, layer.nr_cpus);
         }
 
+        if self.open_metrics_format {
```

Comment:
Is separating out if/else into separate if blocks intentional? Would it be easier to read if the `info!` logging is moved into its own function?

Reply:
Hmm, it's just a control flow difference - info! logging happens in every iteration of the loop whereas OM logging happens only once.

The OM approach is to set all the metrics and then output them all in one shot. I could make the info! logging work similarly but I thought that would look more ugly since I'd need to get() all the OM metrics again.